### PR TITLE
Remove Additional Dict Causing TypeError for Llama Models

### LINKED ISSUE #347: Conversion Lit GPT (Llama-2-7b-chat-hf) to original model
Hi! Thanks for the new commits on the issue #340 . I ran the code as you suggested, but got this error.

`TypeError: copy_weights_llama() got multiple values for argument 'saver'
`

I tried with another checkpoint files but it is still the same. Please let me know what I should do. Thank you!

### PR DESCRIPTION
**Pull request by JustinGoheen**
- removed extra dict causing TypeError
- added appropriate test
- also applied formatting to test file

fixes #347

### CODE REVIEW

File: scripts/convert_lit_checkpoint.py [hunk 1]
```diff
@@ -210,8 +210,7 @@ def convert_lit_checkpoint(
     if "falcon" in model_name:
         copy_fn = partial(copy_weights_falcon, "40b" if config.n_embd == 8192 else "7b")
     elif config._mlp_class == "LLaMAMLP":
-        qkv_weights = {}
```
Comment (JustinGoheen) [inline scripts/convert_lit_checkpoint.py:213]: this dict is not needed

File: tests/test_convert_lit_checkpoint.py [hunk 2]
```diff
@@ -14,24 +14,46 @@ def test_convert_lit_checkpoint(tmp_path):
 
     ckpt_name = "lit_model.pth"
 
-    with pytest.raises(
-        RuntimeError, match="open file failed because of errno 2 on fopen"
-    ):
-        convert_lit_checkpoint(
-            checkpoint_name=ckpt_name, checkpoint_dir=tmp_path, model_name="falcon-7b"
-        )
+    with pytest.raises(RuntimeError, match="open file failed because of errno 2 on fopen"):
```
Comment (JustinGoheen) [inline tests/test_convert_lit_checkpoint.py:17]: updated formatting

File: tests/test_convert_lit_checkpoint.py [hunk 3]
```diff
@@ -14,24 +14,46 @@ def test_convert_lit_checkpoint(tmp_path):
...
+        convert_lit_checkpoint(checkpoint_name=ckpt_name, checkpoint_dir=tmp_path, model_name="falcon-7b")
```
Comment (JustinGoheen) [inline tests/test_convert_lit_checkpoint.py:18]: updated formatting

File: tests/test_convert_lit_checkpoint.py [hunk 4]
```diff
@@ -14,24 +14,46 @@ def test_convert_lit_checkpoint(tmp_path):
...
 
     ckpt_path = tmp_path / "lit_model.pth"
     ckpt_path.touch()
     with mock.patch("scripts.convert_lit_checkpoint.lazy_load") as load:
-        convert_lit_checkpoint(
-            checkpoint_name=ckpt_name, checkpoint_dir=tmp_path, model_name="falcon-7b"
-        )
+        convert_lit_checkpoint(checkpoint_name=ckpt_name, checkpoint_dir=tmp_path, model_name="falcon-7b")
```
Comment (JustinGoheen) [inline tests/test_convert_lit_checkpoint.py:23]: updated formatting

File: tests/test_convert_lit_checkpoint.py [hunk 5]
```diff
@@ -14,24 +14,46 @@ def test_convert_lit_checkpoint(tmp_path):
...
     load.assert_called_with(ckpt_path)
 
     assert {p.name for p in tmp_path.glob("*")} == {"lit_model.pth", "lit_model.bin"}
 
 
+def test_convert_lit_checkpoint__llama2(tmp_path):
```
Comment (JustinGoheen) [inline tests/test_convert_lit_checkpoint.py:29]: new test to check for TypeError

File: tests/test_convert_lit_checkpoint.py [hunk 6]
```diff
@@ -127,9 +147,7 @@ def test_against_original_llama2(size):
         ours_kwargs = {"name": "Llama-2-70b-chat-hf", "n_query_groups": 2}
         theirs_kwargs = {"num_key_value_heads": 2}
 
-    ours_config = Config.from_name(
-        n_layer=2, n_head=8, n_embd=32, intermediate_size=86, **ours_kwargs
-    )
+    ours_config = Config.from_name(n_layer=2, n_head=8, n_embd=32, intermediate_size=86, **ours_kwargs)
```
Comment (JustinGoheen) [inline tests/test_convert_lit_checkpoint.py:150]: updated formatting

File: tests/test_convert_lit_checkpoint.py [hunk 7]
```diff
@@ -188,19 +206,13 @@ def test_maybe_unwrap_state_dict(tmp_path):
 
     # convert and check that model key does not exist
     # and that a known key for pythia exists
-    convert_lit_checkpoint(
-        checkpoint_name=ckpt_name, checkpoint_dir=tmp_path, model_name=model_name
-    )
+    convert_lit_checkpoint(checkpoint_name=ckpt_name, checkpoint_dir=tmp_path, model_name=model_name)
```
Comment (JustinGoheen) [inline tests/test_convert_lit_checkpoint.py:209]: updated formatting

File: tests/test_convert_lit_checkpoint.py [hunk 8]
```diff
@@ -188,19 +206,13 @@ def test_maybe_unwrap_state_dict(tmp_path):
...
     assert ckpt_from_unwrapped.get("embed_out.weight") is not None
 
     # assert maybe_unwrap_state_dict is called
-    with mock.patch(
-        "scripts.convert_lit_checkpoint.maybe_unwrap_state_dict"
-    ) as maybe_unwrap:
-        convert_lit_checkpoint(
-            checkpoint_name=ckpt_name, checkpoint_dir=tmp_path, model_name=model_name
-        )
+    with mock.patch("scripts.convert_lit_checkpoint.maybe_unwrap_state_dict") as maybe_unwrap:
```
Comment (JustinGoheen) [inline tests/test_convert_lit_checkpoint.py:216]: updated formatting

**Review by awaelchli — APPROVED** [review]
Thanks @JustinGoheen 

Just ran this now to confirm
```
python scripts/download.py --repo_id meta-llama/Llama-2-7b-chat-hf --token TOKEN
python scripts/convert_hf_checkpoint.py --checkpoint_dir checkpoints/meta-llama/Llama-2-7b-chat-hf/
python scripts/convert_lit_checkpoint.py --checkpoint_dir checkpoints/meta-l
lama/Llama-2-7b-chat-hf  --checkpoint_name=lit_model.pth
```

**Pull request merged**